internal/common/tree.go

```diff
@@ -60,29 +99,45 @@ func resolveStops(node *DNSTreeNode, label, value, stopLabel string) {
 
 	//remove label from stop labelled children
 	for _, c := range node.Children {
-		d := findDataSetForChild(node, c.Name)
-		//has label and stop label = remove label from this dataset and all children
-		if d != nil && d.Labels[stopLabel] != "" && d.Labels[label] != "" {
-			delete(d.Labels, label)
-			RemoveLabelFromTree(c, label)
-			RemoveLabelFromParents(node, label)
-		} else {
+		dsets := findEndpointsForChild(node, c.Name)
+		if len(dsets) == 0 {
 			resolveStops(c, label, value, stopLabel)
```

Comment:
The "it could be easier" bit (ignorable):
If there are no datasets on a node, will it ever have children? Is this jut an obscure way to `return`?

Reply:
actually this whole check isn't required. I'll remove it